crates/oxc_linter/src/rules/nextjs/no_document_import_in_page.rs

```diff
@@ -0,0 +1,219 @@
+            return;
+        }
+
+        ctx.diagnostic(NoDocumentImportInPageDiagnostic(import_decl.span));
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+    use std::path::PathBuf;
+
+    let pass = vec![
+        (
+            r#"import Document from "next/document"
+			
+			export default class MyDocument extends Document {
+			  render() {
+				return (
+				  <Html>
+				  </Html>
+				);
+			  }
+			}
+			"#,
+            None,
+            None,
+            Some(PathBuf::from("pages/_document.tsx")),
+        ),
+        (
+            r#"import Document from "next/document"
+			
+				export default class MyDocument extends Document {
+				render() {
+					return (
+					<Html>
+					</Html>
+					);
+				}
+				}
+			"#,
+            None,
+            None,
+            Some(PathBuf::from("pages/_document.tsx")),
+        ),
+        (
+            r#"import NextDocument from "next/document"
+
+        	    export default class MyDocument extends NextDocument {
+        	      render() {
+        	        return (
+        	          <Html>
+        	          </Html>
+        	        );
+        	      }
+        	    }
+        	"#,
+            None,
+            None,
+            Some(PathBuf::from("pages/_document.tsx")),
+        ),
+        (
+            r#"import Document from "next/document"
+
+        	    export default class MyDocument extends Document {
+        	      render() {
+        	        return (
+        	          <Html>
+        	          </Html>
+        	        );
+        	      }
+        	    }
+        	"#,
+            None,
+            None,
+            Some(PathBuf::from("pages/_document.tsx")),
+        ),
+        (
+            r#"import NDocument from "next/document"
+
+        	    export default class Document extends NDocument {
+        	      render() {
+        	        return (
+        	          <Html>
+        	          </Html>
+        	        );
+        	      }
+        	    }
+        	"#,
+            None,
+            None,
+            Some(PathBuf::from("pages/_document.tsx")),
+        ),
+        (
+            r#"import NDocument from "next/document"
+
+        	    export default class Document extends NDocument {
+        	      render() {
+        	        return (
+        	          <Html>
+        	          </Html>
+        	        );
+        	      }
+        	    }
+        	"#,
+            None,
+            None,
+            Some(PathBuf::from("pages/_document.tsx")),
+        ),
+        (
+            r#"import Document from "next/document"
+
+        	    export default class MyDocument extends Document {
+        	      render() {
+        	        return (
+        	          <Html>
+        	          </Html>
+        	        );
+        	      }
+        	    }
+        	"#,
+            None,
+            None,
+            Some(PathBuf::from("src/pages/_document.tsx")),
```

Comment:
some pass cases are missing - 

https://github.com/vercel/next.js/blob/61803f818a1fd2865e481719ef196ad3b4180160/test/unit/eslint-plugin-next/no-document-import-in-page.test.ts#L61C1-L89C6

e.g. `pages/_document/index.js',`/ and `pages/_document.page.tsx'`

Reply:
added it!